client/src/components/ActiveChat/Input.js

```diff
@@ -22,7 +23,23 @@ const Input = (props) => {
   const [text, setText] = useState("");
   const { postMessage, otherUser, conversationId, user } = props;
 
+  const emitTypingStatus = (otherUserIsTyping) => {
+    socket.emit("set-typing-status", {
+      conversationId,
+      otherUserIsTyping,
+    });
+  };
```

Comment:
Curious on your thoughts on throttling this? If say we're typing really fast would this socket be emitting every single character typed? If we throttle we could reduce the number of emits being fired?

Probably not a huge deal for the scale of this application but if it were to have lots and lots of users typing at once it could lead to a lot of bandwidth being used.

Reply:
I think that maybe there could be a boolean property on the conversation state such as `userIsTyping`. I could then refactor `emitTypingStatus()` to only emit the socket event if `userIsTyping` is not the same as what is passed in to the function. After the `if` block, I would set `userIsTyping` to whatever was passed to `emitTypingStatus()` as the argument.

Though I'm not sure if `userIsTyping` should be in the redux store or kept track locally in the Input component.

Reply:
Interesting though, I do think for simplicity implementing a simple throttle or debounce would suffice for it. Though, not super important but just wanted to give you a heads up. 👍🏻 

Let's say for example the user types 150 characters, for every character we would fire a socket event. With throttling that could be reduced down to like 20 ish events or even less depending on the throttle delay set. Overall though, I like your implementation and it's clean and easy to maintain and add something like that in! 

You're good to merge this in Michael